capa explorer: unit test fails

capa v4.0.1

```
FAIL mimikatz-function=0x4011FB-offset(-0x1)-True
Traceback (most recent call last):
  File "C:/Exclusions/capa-4.0.1/tests/test_ida_features.py", line 61, in test_ida_features
    fixtures.do_test_feature_presence(get_ida_extractor, sample, scope, feature, expected)
  File "C:\Exclusions/capa-4.0.1/tests\fixtures.py", line 792, in do_test_feature_presence
    assert feature.evaluate(features) == expected, msg
AssertionError: offset(-0x1) should be found in function=0x4011FB
FAIL mimikatz-function=0x4011FB-offset(-0x2)-True
Traceback (most recent call last):
  File "C:/Exclusions/capa-4.0.1/tests/test_ida_features.py", line 61, in test_ida_features
    fixtures.do_test_feature_presence(get_ida_extractor, sample, scope, feature, expected)
  File "C:\Exclusions/capa-4.0.1/tests\fixtures.py", line 792, in do_test_feature_presence
    assert feature.evaluate(features) == expected, msg
AssertionError: offset(-0x2) should be found in function=0x4011FB
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -64,6 +64,7 @@
 - fix: accept only plaintext pasted content #1194 @williballenthin
 - fix: UnboundLocalError #1217 @williballenthin
 - extractor: add support for COFF files and extern functions #1223 @mike-hunhoff
+- fix: assume 32-bit displacement for offsets #1250 @mike-hunhoff
 
 ### Development
 

--- a/capa/features/extractors/ida/helpers.py
+++ b/capa/features/extractors/ida/helpers.py
@@ -221,7 +221,8 @@ def get_op_phrase_info(op: idaapi.op_t) -> Dict:
         return {}
 
     scale = 1 << ((op.specflag2 & 0xC0) >> 6)
-    offset = op.addr
+    # IDA ea_t may be 32- or 64-bit; we assume displacement can only be 32-bit
+    offset = op.addr & 0xFFFFFFFF
 
     if op.specflag1 == 0:
         index = None

--- a/capa/features/extractors/ida/insn.py
+++ b/capa/features/extractors/ida/insn.py
@@ -213,7 +213,11 @@ def extract_insn_offset_features(
             continue
 
         p_info = capa.features.extractors.ida.helpers.get_op_phrase_info(op)
-        op_off = p_info.get("offset", 0)
+
+        op_off = p_info.get("offset", None)
+        if op_off is None:
+            continue
+
         if idaapi.is_mapped(op_off):
             # Ignore:
             #   mov esi, dword_1005B148[esi]

